src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/MigrationSessiontTemplateMatcher.kt

```diff
@@ -188,6 +187,32 @@ class MigrationSessionTemplateMatcher(
     return bestMatch
   }
 
+  private fun logMatchDetails(
+    migrateMatch: MigrateMatch,
+    it: SessionTemplate,
+    prisonCode: String,
+    prisonerId: String,
+    prisonLevelMap: Map<PrisonerHousingLevels, String?>,
+  ) {
+    with(migrateMatch) {
+      val builder = StringBuilder()
+      builder.append("getNearestSessionTemplate session is not permitted ")
+      builder.append("ref:${it.reference}/$prisonCode/$prisonerId ")
+      builder.append("locationScore:$locationScore")
+      builder.append("category:$category ")
+      builder.append("enhanced:$enhanced ")
+      builder.append("timeProximity:$timeProximity ")
+      builder.append("roomMatch:$roomNameMatch ")
+      builder.append("dateProximity:$validFromDateProximityDays ")
+      builder.append("location:${locationToString(prisonLevelMap)} ")
+      LOG.debug(builder.toString())
+    }
+  }
+
+  private fun locationToString(prisonLevelMap: Map<PrisonerHousingLevels, String?>): String {
+    return prisonLevelMap.values.filterNotNull().toString()
```

Comment:
will it be better to concat them with "-"?

Reply:
I will ignore looks good already